### ISSUE TITLE
LIFEFORCE (JP),

### ISSUE THREAD

**Issue by marumusisan** [MiSTer-devel/Arcade-Salamander_MiSTer#3]
In LIFEFORCE (JP), the C button on the 2 Player side cannot be set.

**Comment by marumusisan** [comment]
Another issue is that the credit button sometimes doesn't work on LIFEFORCE(JP).
The exact conditions are unknown, so we will investigate and let you know as soon as we know.

**Comment by ika-musume** [comment]
Currently, you need to create a button setup for the Life Force (JP) - that PCB uses 3 buttons, but Konami has expanded the inputs from 2 buttons and has not maintained compatibility. I'm thinking of modifying the mra to give the core a setup bit so that the core can automatically remap the buttons.

I don't know about the credit button not working, I've never encountered that issue.

**Comment by marumusisan** [comment]
It seems that when you set the credit button on the 2 player side, the credit button stops working.

This is a request, but could you please make a prize version of Salamander (SP CLONE Ver.)
?
https://www.youtube.com/watch?v=_JqGGJ_-9fw

**Comment by ika-musume** [comment]
Will check what's wrong. This will be fixed on the next release.

That ROM is not listed on MAME, so I need the ROM to add support of that version.

**Comment by marumusisan** [comment]
Salamander (sp clone) is supported by HBMAME

**Comment by marumusisan** [comment]
https://xgf.nu/7jcfg

Spclone is here.
If you are asked for a password
Password:spc

**Comment by ika-musume** [comment]
I've downloaded both spcloneo and spclone, I'll fix the button input issue and add new mras.

**Comment by marumusisan** [comment]
Additional problems.
The settings made with the DIP SWITCH are not reflected.
The sound made when defeating the boss on stage 3 is strange.
These are the two problems.
Both were confirmed with Salamander.

**Comment by ika-musume** [comment]
If you have changed the DIP switch settings, you will need to reload the core. The MiSTer framework does not seem to send updated switch information back to the core after a soft reset. 

That sound is generated by K007232, and there is currently one issue I have found. I don't know if this affects that, but I'll check it out.

**Comment by mhmetalheart** [comment]
hello,
I just made the spclone & spcloneo HBmame by myself. It was working. And just 'add' another botton in the mra, That will totally work for 1P.
But issue here: for example, I use 8bitdo SN30 pro controller (same map as SFC): buttons default="Select,L,Start,A,Y,X" names="Coin,Start,Service,Shoot,Missile,Upgrade"
sel= coin. Correct.
L=start. nothing happened.
Start=service. Wrong, actually, it was start NOW.
A=shoot. wrong. Y is shoot. 
Y=missile. wrong. X is missile.
X=upgrade. wrong. B is upgrade.

And yes, for 2P, the 3rd button doesn't work.

Please resolve.

**Comment by ika-musume** [comment]
Replies inline

In LIFEFORCE (JP), the C button on the 2 Player side cannot be set.
**>>fixed, a problem with the code**

It seems that when you set the credit button on the 2 player side, the credit button stops working.
**>>I don't have a way to test the second player input, so I can't confirm, but the coin signal goes to the board core just fine.**

This is a request, but could you please make a prize version of Salamander (SP CLONE Ver.)
**>>spclone/spcloneo added**

The sound made when defeating the boss on stage 3 is strange.
**>>Both Furrtek and my implementation of the K007232 make strange sounds. Furrtek's implementation has a slightly faster playback speed for the 2nd channel, but I don't know if this is a design flaw with Konami or a bug in his schematic. My implementation has the same playback speed for all PCM channels, but it sounds weird here too. I checked with ILA and they are not phase-canceled, so we should hear the sound. I think the MiSTer framework is blocking the sound somehow, I can't fix it right away, so I'll leave it as a long-term task.**

But issue here: for example, I use 8bitdo SN30 pro controller (same map as SFC): buttons default="Select,L,Start,A,Y,X"...
**>>Fixed to allow the core to receive the number of buttons to use from the MRA. Try again with a new RBF and MRA**

**Comment by mhmetalheart** [comment]
confirm buttons issue resolved. Confirm 2P can use 3rd button also.

**Comment by marumusisan** [comment]
thank you I'm out of town now, so I'll check when I get home!

2024年10月31日(木) 12:53 mhmetalheart ***@***.***>:

**Comment by marumusisan** [comment]
I ran update_all.sh to update the mra,
but the sp clone mra was not added.
Also, when I configured the buttons in Life Force (JP),
only the shot button worked, and power-ups and missiles were unusable.
I am using 8bitdo's M30Pro with a Bluetooth connection.
Also, the default is 70,000 points, and then it extends to 200,000 points every time, but with the MiSTer core it is fixed at only 100,000 points.
